pandas/io/formats/format.py

```diff
@@ -1794,12 +1794,12 @@ def _format_datetime64_dateonly(
 
 
 def get_format_datetime64(
-    is_dates_only: bool, nat_rep: str = "NaT", date_format: str | None = None
+    is_dates_only_result: bool, nat_rep: str = "NaT", date_format: str | None = None
```

Comment:
ideally, let's not rename function arguments

Reply:
Agreed this is not ideal- not sure how to account for the issue that is_dates_only is both a function name and an argument in a different function, which seems to be what's causing the issue.  Would it be better to rename the function rather than then argument?

Reply:
ah, I see what you mean, thanks. gosh, that's dirty 😄 taking a look

Reply:
The function `is_dates_only` is only used within this file - how about we just rename it to `_is_dates_only`?

Reply:
> The function is_dates_only is only used within this file

sorry I was wrong, it's used inside that other function in fact

Reply:
In 

https://github.com/pandas-dev/pandas/blob/694c0cd9763710f7576ba893ae4cb6a410007e2a/pandas/io/formats/style_render.py#L1806

the argument `slice` is called `slice_` to avoid shadowing the builtin `slice` - shall we do the same here?

If you then remove `format.py` from the `.pre-commit-config.yaml` file, the rest looks good 👍